docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/DockerContainerDefinitionParser.java

```diff
@@ -98,36 +101,113 @@ public static DockerCompositions convert(String content, DefinitionFormat defini
         URI defaultUri = null;
-        try {
             switch (definitionFormat) {
                 case COMPOSE: {
-                    final URL resource = DockerContainerDefinitionParser.class.getResource("/" + DEFAULT_DOCKER_COMPOSE_DEFINITION_FILE);
-                    if (resource != null) {
-                        defaultUri = resource.toURI();
-                    }
+                    defaultUri = getDefaultFileLocation(DEFAULT_DOCKER_COMPOSE_DEFINITION_FILE);
                     break;
                 }
                 case CUBE: {
-                    final URL resource = DockerContainerDefinitionParser.class.getResource("/" + DEFAULT_CUBE_DEFINITION_FILE);
-                    if (resource != null) {
-                        defaultUri = resource.toURI();
-                    }
+                    defaultUri = getDefaultFileLocation(DEFAULT_CUBE_DEFINITION_FILE);
                     break;
                 }
                 default: {
-                    final URL resource = DockerContainerDefinitionParser.class.getResource("/" + DEFAULT_CUBE_DEFINITION_FILE);
-                    if (resource != null) {
-                        defaultUri = resource.toURI();
-                    }
+                    defaultUri = getDefaultFileLocation(DEFAULT_DOCKER_COMPOSE_DEFINITION_FILE);
                 }
             }
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException(e);
-        }
         if (defaultUri == null) {
             logger.fine("No Docker container definitions has been found. Probably you have defined some Containers using Container Object pattern and @Cube annotation");
             return new DockerCompositions();
         }
-        return convert(Paths.get(defaultUri), definitionFormat);
+        return convert(defaultUri, definitionFormat);
     }
+
+    private static URI getDefaultFileLocation(String filename) {
+
+        // src/{test/main}/docker
+        URI docker = checkSrcTestAndMainResources(filename, "docker");
+
+        if (docker == null) {
+
+            // .
+            docker = checkRoot(filename);
+
+            if (docker == null) {
+                // src/distribution
+                docker = checkDistributionDirectory(filename);
+
+                if (docker == null) {
+                    // src/{test, main}/resources/docker
+                    docker = checkSrcTestAndMainResources(filename, "resources/docker");
+
+                    if (docker == null) {
+
+                        // src/{test, main}/resources
+                        docker = checkSrcTestAndMainResources(filename, "resources");
+                    }
+                }
+            }
+        }
+
+        return docker;
+
+    }
+
+    private static URI checkRoot(String filename) {
+        final Path rootPath = Paths.get(filename);
+        Path finalPath;
+
+        if ((finalPath = exists(rootPath)) != null) {
```

Comment:
I dont really like this approach of assignment within an `if` statement. Makes it somehow blurry and reminds me of my ANSI C times ;)